bridge-fungible-token/src/utils.sw

```diff
@@ -150,28 +150,20 @@ pub fn decompose(val: b256) -> (u64, u64, u64, u64) {
 
 /// Read the bytes passed as message data into an in-memory representation using the MessageData type.
 pub fn parse_message_data(msg_idx: u8) -> MessageData {
+    let token: b256 = input_message_data(msg_idx, 32).into();
+
     let mut msg_data = MessageData {
-        token: ZERO_B256,
+        token,
         from: ZERO_B256,
         to: Identity::Address(Address::from(ZERO_B256)),
         amount: ZERO_B256,
         len: 160,
     };
 
     // Parse the message data
-    let token = input_message_data(msg_idx, 32);
-    let ptr = __addr_of(msg_data.token);
-    token.buf.ptr().copy_to::<b256>(ptr, 1);
-    // TODO: when https://github.com/FuelLabs/sway/issues/4450 is fixed, remove 3 lines above and uncomment line below. Same applies to msg_data.from, msg_data.amount & msg_data.to
-    // msg_data.token = input_message_data(msg_idx, 32).into();
-    let from = input_message_data(msg_idx, 32 + 32);
-    let ptr_2 = __addr_of(msg_data.from);
-    from.buf.ptr().copy_to::<b256>(ptr_2, 1);
-    // msg_data.from = input_message_data(msg_idx, 32 + 32).into();
-    let amount = input_message_data(msg_idx, 32 + 32 + 32 + 32);
-    let ptr_3 = __addr_of(msg_data.amount);
-    amount.buf.ptr().copy_to::<b256>(ptr_3, 1);
-    // msg_data.amount = input_message_data(msg_idx, 32 + 32 + 32 + 32).into();
+    // TODO: Bug, have to mutate this struct for these values or tests fail
```

Comment:
Do we still need this part? 
<img width="603" alt="image" src="https://github.com/FuelLabs/bridge-fungible-token/assets/18346821/6c46f8fb-5117-4344-bf8c-638060b6d012">